api/s2n.h

```diff
@@ -73,15 +76,14 @@ extern int s2n_config_add_cert_chain_and_key_with_extensions(struct s2n_config *
         const char *private_key_pem, s2n_tls_extension *extensions, uint16_t num_extensions);
```

Comment:
Do we still need `s2n_config_add_cert_chain_and_key_with_extensions` if we have `s2n_config_set_extensions`? I don't think it's been in the API long enough to have backward-compatibility concerns.